app/gui/src/providers/featureFlags.ts

```diff
@@ -85,14 +72,21 @@ export const flagsStore = createStore<FeatureFlagsStore>()(
       name: 'enso-feature-flags',
       version: 1,
       merge: (persistedState, newState) => {
-        /**
-         * Mutates the state with provided feature flags
-         */
-        function unsafeMutateFeatureFlags(flags: Partial<FeatureFlags>) {
-          unsafeWriteValue(newState, 'featureFlags', {
-            ...newState.featureFlags,
-            ...flags,
-          })
+        /** Mutates the state with provided feature flags. */
+        function unsafeMutateFeatureFlags(flags: {
+          [K in keyof FeatureFlags]?: FeatureFlags[K] | undefined
+        }) {
+          const newFeatureFlags = { ...newState.featureFlags }
+          for (const kv of unsafeEntries(flags)) {
+            if (!kv) {
+              continue
+            }
```

Comment:
`unsafeEntries` cannot return undefined element, can it?

Reply:
nice catch :) fixed type definitions